CMakeLists.txt

```diff
@@ -14,7 +14,7 @@ message(STATUS "\n\n====== Finding 3rd Party Packages ======\n")
 find_package(ament_cmake REQUIRED)
 find_package(ament_cmake_python REQUIRED)
 find_package(maliput REQUIRED)
-find_package(pybind11 REQUIRED)
+find_package(pybind11 2.4.3 REQUIRED)
```

Comment:
Adding a suggested version (Note: It isn't `EXACT REQUIRED`) given that when working with the entire workspace,` delphyne` would still need the custom pybind11 version which matches (`2.6.x`). Therefore, to avoid picking up the built-from-source one, a suggestion on the version here might be required.
@agalbachicar

Reply:
> What would it happen if the version is bumped upstream?

See your point. We could use just `2.4` instead of `2.4.3` and at least if it's a `patch` it will still match. 
However it is just a suggestion, if you don't have 2.4.3 but you have other system version installed it will find it, unless of course you are compiling it in your workspace. 
It isn't 100% ideal, however dealing with two different versions installed in your system isn't a common thing

Reply:
As per https://github.com/maliput/maliput_infrastructure/issues/225#issuecomment-1156763365, pybind11 won't be part of the workspace so no need to use a particular version at all